### PR TITLE
NavBar migrated to ts

### PR DESCRIPTION
**Pull request by Muhindo-Galien**
# PR Description

Please include a brief description of the pull request you want to raise.

## Description of tasks that were expected to be completed

List all the tasks and sub-tasks that were assigned to you Make sure you check every completed task. list them in this style

- [ ] Create .....
- [ ] Add .....
- [ ] ......

## Description of tasks that were expected to be completed and that are not yet completed

- [ ] Create .....
- [ ] Add .....
- [ ] ......

## How this may be tested?

Please describe the tests that you ran to verify your functionalities. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

1. run `npm i` to install all required dependencies
2. run `npm run dev`
3. step 3
4. step 4...

## Screenshots (If appropriate/applicable)

Provide screenshots related to your PR in case it's appropriate and applicable

## Please check this Checklist before you submit your PR:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] My code generates no warnings
- [ ] There are no vulnerabilities

### CODE REVIEW

File: components/AddNewSongPage/index.jsx [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-// export default AddNewSongPage;
+const AddNewSongPage = () => {
+  const [isTopNav, setIsTopNav] = useState(false);
+  return (
+    <PageCard setIsTopNav={setIsTopNav}>
+      <AddNewSinger isTopNav={isTopNav} />
+    </PageCard>
+  );
+};
+export default AddNewSongPage;
```
Comment (Jonath-z) [inline components/AddNewSongPage/index.jsx:14]: this file still a `jsx` extension, it's should also be migrated or commented

File: components/BottomPaler/BottomPalyer.jsx [hunk 2]
```diff
@@ -1,101 +1,101 @@
...
 
-// BottomPlayer.propTypes = {
-//   isFixed: PropTypes.bool,
-// };
+BottomPlayer.propTypes = {
+  isFixed: PropTypes.bool,
+};
 
-// export default BottomPlayer;
+export default BottomPlayer;
```
Comment (Jonath-z) [inline components/BottomPaler/BottomPalyer.jsx:101]: This file is `jsx` extension as well

File: components/lib/atoms/index.js [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
-//   key: "displaysettingsModalAtom",
-//   default: false,
-// });
+const displaySettingsModalAtom = atom({
+  key: "displaysettingsModalAtom",
+  default: false,
+});
 
-// export { openMenuAtom, leftNavPinedAtom, displaySettingsModalAtom };
+export { openMenuAtom, leftNavPinedAtom, displaySettingsModalAtom };
```
Comment (Jonath-z) [inline components/lib/atoms/index.js:20]: jsx components should be commented before creating the PR

File: components/modules/NavBar/LeftNav/DashBoard/index.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        <DashboardMenu />
+      </div>
+    </div>
+  );
+};
+
+DashBord.propTypes = {
+  isDashboardSubmenu: PropTypes.bool.isRequired,
+  toggleSubmenu: PropTypes.func.isRequired,
+};
```
Comment (Jonath-z) [inline components/modules/NavBar/LeftNav/DashBoard/index.tsx:38]: for typescript files we don't need `propType` anymre

File: components/modules/NavBar/LeftNav/DashBoard/index.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { ChevronRightIcon, HomeIcon } from "../../../__modules/Vectors";
+import DashboardMenu from "./DashboardMenu";
+import PropTypes from "prop-types";
+
+const DashBord = ({ isDashboardSubmenu, toggleSubmenu }) => {
```
Comment (Jonath-z) [inline components/modules/NavBar/LeftNav/DashBoard/index.tsx:6]: You should add the type of those props

File: components/modules/NavBar/LeftNav/AdminMenu/index.tsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { AdminIcon, ChevronRightIcon } from "../../../__modules/Vectors";
+import AdminSubmenu from "./AdminSubmenu";
+import PropTypes from "prop-types";
+
+const AdminMenu = ({ isAdminSubmenu, toggleSubmenu }) => {
```
Comment (Jonath-z) [inline components/modules/NavBar/LeftNav/AdminMenu/index.tsx:6]: As this is a ts  file already, we should a type or interface for those props

File: components/modules/NavBar/TopNav/index.tsx [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+import { openMenuAtom } from "../../../lib/atoms";
+import { useRecoilState } from "recoil";
+import NotificationModal from "./__modules/NotificationModal";
+import ProfileMenuModal from "./__modules/ProfileMenuModal";
+import PropTypes from "prop-types";
+import Link from "next/link";
+import { ChevronRightIcon, LoginIcon, MenuOutlineIcon, MicrophoneIcon, NotificationIcon, SeachIcon } from "../../__modules/Vectors";
+
+const TopNav = ({ setIsTopNav }) => {
```
Comment (Jonath-z) [inline components/modules/NavBar/TopNav/index.tsx:11]: Add a type on this prop as well

File: components/modules/NavBar/TopNav/index.tsx [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+        <ProfileMenuModal />
+      </div>
+    </>
+  );
+};
+
+TopNav.propstype = {
+  setIsTopNav: PropTypes.func,
+};
+
```
Comment (Jonath-z) [inline components/modules/NavBar/TopNav/index.tsx:136]: we don't need propsType anymore in this file

File: components/modules/__modules__/Card/PageCard.tsx [hunk 9]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import { useRecoilValue } from "recoil";
+import PropTypes from "prop-types";
+import BottomPlayer from "../../../BottomPaler/BottomPalyer";
+import { leftNavPinedAtom } from "../../../lib/atoms";
+import { LeftNav, TopNav } from "../../NavBar";
+
+const PageCard = ({setIsTopNav,children}) => {
```
Comment (Jonath-z) [inline components/modules/__modules__/Card/PageCard.tsx:8]: Type for this prop as well

File: components/modules/__modules__/Card/PageCard.tsx [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+      <LeftNav />
+      <BottomPlayer />
+    </>
+  );
+};
+
+PageCard.propTypes = {
+  children: PropTypes.element.isRequired,
+  setIsTopNav: PropTypes.func,
+};
```
Comment (Jonath-z) [inline components/modules/__modules__/Card/PageCard.tsx:27]: no need of proptype

File: components/modules/__modules__/_Vectors/index.jsx [hunk 11]
```diff
@@ -1,9 +1,9 @@
...
-// export { VHamburger, VDots, VWave, VPrev, VNext, VPlay, VPause, VStar };
+import { GiHamburgerMenu as VHamburger } from 'react-icons/gi';
+import { BiDotsVerticalRounded as VDots } from "react-icons/bi"
+import { FaPlay as VPlay } from 'react-icons/fa'
+import { BsPauseCircleFill as VPause } from 'react-icons/bs'
+import { IoStarOutline as VStar } from 'react-icons/io5'
+import { ImNext as VNext } from 'react-icons/im';
+import { ImPrevious as VPrev } from 'react-icons/im'
+import VWave from './wavePlayVector';
+export { VHamburger, VDots, VWave, VPrev, VNext, VPlay, VPause, VStar };
```
Comment (Jonath-z) [inline components/modules/__modules__/_Vectors/index.jsx:9]: js file should be commented

File: components/modules/NavBar/LeftNav/AdminMenu/index.tsx [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+      </div>
+    </div>
+  );
+};
+
+AdminMenu.propTypes = {
+  isAdminSubmenu: PropTypes.bool.isRequired,
+  toggleSubmenu: PropTypes.func.isRequired,
+};
+
```
Comment (Jonath-z) [inline components/modules/NavBar/LeftNav/AdminMenu/index.tsx:39]: no need of propType here

**Review by Jonath-z — CHANGES_REQUESTED** [review]

File: components/modules/NavBar/TopNav/index.tsx (hunk 8 above)
Comment (Muhindo-Galien) [reply components/modules/NavBar/TopNav/index.tsx:136]: Got it... lemme work on it

File: components/modules/__modules__/_Vectors/index.jsx (hunk 11 above)
Comment (Muhindo-Galien) [reply components/modules/__modules__/_Vectors/index.jsx:9]: got it bro

File: components/lib/atoms/index.js (hunk 3 above)
Comment (Muhindo-Galien) [reply components/lib/atoms/index.js:20]: i have to use this file for the NavBar component

File: components/modules/NavBar/LeftNav/DashBoard/index.tsx (hunk 5 above)
Comment (Muhindo-Galien) [reply components/modules/NavBar/LeftNav/DashBoard/index.tsx:6]: Added

File: components/modules/NavBar/LeftNav/AdminMenu/index.tsx (hunk 12 above)
Comment (Muhindo-Galien) [reply components/modules/NavBar/LeftNav/AdminMenu/index.tsx]: Every suggested change has been made already!
feel free to check out everything about my changes now

**Review by Jonath-z — APPROVED** [review]

**Pull request merged**